paasta_tools/tron_tools.py

```diff
@@ -0,0 +1,379 @@
+        except KeyError:
+            raise TronNotConfigured('Could not find default PaaSTA cluster in system Tron config')
+
+    def get_url(self):
+        """:returns The URL for the Tron master's API"""
+        try:
+            return self['url']
+        except KeyError:
+            raise TronNotConfigured('Could not find URL of Tron master in system Tron config')
+
+
+def load_tron_config():
+    return TronConfig(load_system_paasta_config().get_tron_config())
+
+
+class TronClient:
+
+    def __init__(self, url):
+        self.master_url = url
+
+    def _request(self, method, url, data):
+        headers = {'User-Agent': get_user_agent()}
+        kwargs = {
+            'url': urljoin(self.master_url, url),
+            'headers': headers,
+        }
+        if method == 'GET':
+            kwargs['params'] = data
+            response = requests.get(**kwargs)
+        elif method == 'POST':
+            kwargs['data'] = data
+            response = requests.post(**kwargs)
+        else:
+            raise ValueError('Unrecognized method: {}'.format(method))
+
+        # Raise an exception if unsuccessful.
+        response.raise_for_status()
+        return response.json()
+
+    def _get(self, url, data=None):
+        return self._request('GET', url, data)
+
+    def _post(self, url, data=None):
+        return self._request('POST', url, data)
+
+    def update_namespace(self, namespace, new_config):
+        current_config = self._get('/api/config', {'name': namespace})
+        if new_config == current_config['config']:
+            log.info('No change in config, skipping update.')
+            return
+
+        self._post(
+            '/api/config',
+            data={
+                'name': namespace,
+                'config': new_config,
+                'hash': current_config['hash'],
+                'check': 0,
+            },
+        )
+
+    def list_namespaces(self):
+        response = self._get('/api')
+        return response.get('namespaces', [])
+
+
+def get_tron_client():
+    return TronClient(load_tron_config().get_url())
+
+
+class TronActionConfig(InstanceConfig):
+    config_filename_prefix = 'tron'
+
+    def __init__(self, service, instance, config_dict, branch_dict, soa_dir=DEFAULT_SOA_DIR):
+        super(TronActionConfig, self).__init__(
+            cluster=config_dict.get('cluster'),
+            instance=instance,
+            service=service,
+            config_dict=config_dict,
+            branch_dict=branch_dict,
+            soa_dir=soa_dir,
+        )
+        self.job, self.action = instance.split(SPACER)
+
+    def get_job_name(self):
+        return self.job
+
+    def get_action_name(self):
+        return self.action
+
+    def get_deploy_group(self):
+        return self.config_dict.get('deploy_group', '')
+
+    def get_cmd(self):
+        return self.config_dict.get('command')
+
+    def get_executor(self):
+        executor = self.config_dict.get('executor', 'ssh')
```

Comment:
this function will break if we ever introduce another executor, right? how about

```
      return 'mesos' if executor == 'paasta' else executor
```

Reply:
Yep, that looks better.